syft/execution/role.py

```diff
@@ -117,23 +135,31 @@ def execute(self, args_):
     def _instantiate_inputs(self, args_):
         """ Takes input arguments for this role and generate placeholders.
         """
-        input_placeholders = tuple(
+
+        def instantiate_nested_input(input_placeholders, obj):
+            """
+                Function to instantiate all the placeholders in order, even if in nested structures.
+                Args:
+                    input_placeholders: the input placeholders generated at build time.
+                    args: the arguments received on __call__
+            """
+
+            if isinstance(obj, FrameworkTensor):
+                placeholder = input_placeholders.pop(0)
+                placeholder.instantiate(obj)
+            elif isinstance(obj, (list, tuple, set)):
+                for elem in obj:
+                    instantiate_nested_input(input_placeholders, elem)
+            elif isinstance(obj, dict):
+                _ = [
```

Comment:
I don't think you need the `_ = `. And also, maybe a classic for loop would be more readable and easier to understand than a list comprehension here